sql/src/test/java/io/crate/operation/scalar/arithmetic/PowerFunctionTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.scalar.arithmetic;
+
+import io.crate.operation.scalar.AbstractScalarFunctionsTest;
+import org.junit.Test;
+
+public class PowerFunctionTest extends AbstractScalarFunctionsTest {
+
+    @Test
+    public void testPowerWithIntegers() {
+        assertEvaluate("power(2,4)", 16.0);
```

Comment:
I think the multiply operation is slightly different, in that if the both the input values are of an integral type, the result will be of an integral type.

The same doesn't hold for the power function as a negative exponent will yield a decimal type result. If we want to maintain the type of the inputs (integrals) then we will have to round the result - ie: 2^-3=0, 2^-1=1. 
Or in this case, if the result contains a decimal type, shall we change the type of the result to double (this feels non intuitive to me), but keep it integral for the other cases?